### ISSUE TITLE
How to transpose?

### ISSUE THREAD

**Issue by grothesque** [fre-hu/mdarray#8]
In current mdarray, is it possible to perform a (generalized) transpose operation, i.e. change the order of dimensions without copying the data?

Since a strided layout is supported this should be technically possible.

**Comment by fre-hu** [comment]
See the reorder, reorder_mut and into_reordered methods. Are they what you mean?

**Comment by fre-hu** [comment]
There are functions for general permutations also (permute/permut_mut/into_permuted). Like the view functions they are implemented for fixed rank, so you might have issues with dynamic rank. Maybe the interface for all of these should be updated in some way. 

Yes, you can also write a standalone function that uses View::new_unchecked with the pointer and a new mapping from StridedMapping::new. For dynamic rank you don't need any reorganization of types.

**Comment by fre-hu** [comment]
I've made one more commit where the permutation can be dynamic, and also axis arguments in other functions. This simplifies the typical case with dynamic dimensions and is more aligned with ndarray. There are also axis index functions for simple views, which work with dynamic rank. Hopefully this should improve useability.

C++ mdarray does not support dynamic rank.

**Comment by fre-hu** [comment]
Yes I think so, I will close it. I'm not sure what you mean with static arrays, but the permutation must be an IntoShape (i.e Dim tuple, [usize; N], &[usize] or similar).

**Comment by fre-hu** [comment]
I agree, reorder is not so good name.

I was unsure how to define the transposition operation, but it is probably best to follow the convention in Numpy and rename reorder to transpose.

**Comment by fre-hu** [comment]
Fixed, see latest commit.
